src/main/java/com/google/sps/data/GivrUser.java

```diff
@@ -51,53 +56,104 @@ public boolean isMaintainer() {
     return this.isMaintainer;
   }
 
-  //TODO(): get correct moderator status for organization
+  public String getUserEmail() {
+    return this.email;
+  }
+
+  public boolean isLoggedIn() {
+    return this.isLoggedIn;
+  }
+
+  // TODO(): get correct moderator status for organization
   public boolean isModerator(long organizationId) {
       return false;
   }
 
-  public static GivrUser getUserByIdFromDatastore(String userId) {
+  public static boolean checkIfUserWithPropertyExists(String propertyName, String propertyValue) {
+    if (getUserFromDatastoreWithProperty(propertyName, propertyValue) == null) {
+      return false;
+    }
+    return true;
+  }
+
+  // Gets User with propertyName, propertyValue exists within Datastore.
+  public static Entity getUserFromDatastoreWithProperty(String propertyName, String propertyValue) {
```

Comment:
I left this as a public function because it is used in AuthenticateServlet in order to update the User entity.